scripts/deterministic/DeployScroll.s.sol

```diff
@@ -1011,262 +1011,381 @@ contract DeployScroll is DeterminsticDeployment {
-            notnull(L1_STANDARD_ERC20_GATEWAY_PROXY_ADDR),
-            notnull(L2_GATEWAY_ROUTER_PROXY_ADDR),
-            notnull(L2_SCROLL_MESSENGER_PROXY_ADDR),
-            notnull(L2_SCROLL_STANDARD_ERC20_FACTORY_ADDR)
-        );
+        if (getInitializeCount(L2_STANDARD_ERC20_GATEWAY_PROXY_ADDR) == 0) {
+            L2StandardERC20Gateway(L2_STANDARD_ERC20_GATEWAY_PROXY_ADDR).initialize(
+                notnull(L1_STANDARD_ERC20_GATEWAY_PROXY_ADDR),
+                notnull(L2_GATEWAY_ROUTER_PROXY_ADDR),
+                notnull(L2_SCROLL_MESSENGER_PROXY_ADDR),
+                notnull(L2_SCROLL_STANDARD_ERC20_FACTORY_ADDR)
+            );
+        }
     }
 
     function initializeL2WETHGateway() private {
-        L2WETHGateway(payable(L2_WETH_GATEWAY_PROXY_ADDR)).initialize(
-            notnull(L1_WETH_GATEWAY_PROXY_ADDR),
-            notnull(L2_GATEWAY_ROUTER_PROXY_ADDR),
-            notnull(L2_SCROLL_MESSENGER_PROXY_ADDR)
-        );
+        if (getInitializeCount(L2_WETH_GATEWAY_PROXY_ADDR) == 0) {
+            L2WETHGateway(payable(L2_WETH_GATEWAY_PROXY_ADDR)).initialize(
+                notnull(L1_WETH_GATEWAY_PROXY_ADDR),
+                notnull(L2_GATEWAY_ROUTER_PROXY_ADDR),
+                notnull(L2_SCROLL_MESSENGER_PROXY_ADDR)
+            );
+        }
 
         // set WETH gateway in router
         {
             address[] memory _tokens = new address[](1);
             _tokens[0] = notnull(L2_WETH_ADDR);
             address[] memory _gateways = new address[](1);
             _gateways[0] = notnull(L2_WETH_GATEWAY_PROXY_ADDR);
-            L2GatewayRouter(L2_GATEWAY_ROUTER_PROXY_ADDR).setERC20Gateway(_tokens, _gateways);
+            if (L2GatewayRouter(L2_GATEWAY_ROUTER_PROXY_ADDR).ERC20Gateway(_tokens[0]) != _gateways[0]) {
+                L2GatewayRouter(L2_GATEWAY_ROUTER_PROXY_ADDR).setERC20Gateway(_tokens, _gateways);
+            }
         }
     }
 
     function initializeScrollStandardERC20Factory() private {
-        ScrollStandardERC20Factory(L2_SCROLL_STANDARD_ERC20_FACTORY_ADDR).transferOwnership(
+        if (
+            ScrollStandardERC20Factory(L2_SCROLL_STANDARD_ERC20_FACTORY_ADDR).owner() !=
             notnull(L2_STANDARD_ERC20_GATEWAY_PROXY_ADDR)
-        );
+        ) {
+            ScrollStandardERC20Factory(L2_SCROLL_STANDARD_ERC20_FACTORY_ADDR).transferOwnership(
+                L2_STANDARD_ERC20_GATEWAY_PROXY_ADDR
+            );
+        }
     }
 
     function initializeL2Whitelist() private {
         address[] memory accounts = new address[](1);
         accounts[0] = L2_GAS_ORACLE_SENDER_ADDR;
-        Whitelist(L2_WHITELIST_ADDR).updateWhitelistStatus(accounts, true);
+        if (!Whitelist(L2_WHITELIST_ADDR).isSenderAllowed(accounts[0])) {
+            Whitelist(L2_WHITELIST_ADDR).updateWhitelistStatus(accounts, true);
+        }
     }
 
     function transferL2ContractOwnership() private {
```

Comment:
To make sure I understand -- we dropped the early return for `(DEPLOYER_ADDR == OWNER_ADDR)` and so if this isn't the initial deployment, and the original deployment didn't have matching deployer and owner addresses, we will throw an error (reverted tx) since the deployer can't transfer ownership.

Reply:
Yes, we will throw an error (reverted tx) since only owner could transfer ownership.